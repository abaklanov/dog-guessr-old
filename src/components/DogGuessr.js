import React from "react";
import axios from "axios";
import GuessButtons from "./GuessButtons";

class DogGuessr extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      photo: "",
      breedToGuess: ""
    };
  }

  componentDidMount() {
    const response = axios
      .get("https://dog.ceo/api/breeds/image/random")
      .then(
        (resp => {
          this.setState({
            photo: resp.data.message,
            breedToGuess: this.determineBreedToGuess(resp.data.message)
          });
        }).bind(this)
      )
      .catch(err => {
        console.log(err);
      });
  }

  determineBreedToGuess(url) {
    const segments = url.substring(8).split("/");
    return segments[2];
  }

  render() {
    return (
      <div>
        <div className="row justify-content-center no-gutters">
          <div className="col-lg-6 text-center">
            <div id="photo" className=" m-1">
              <img
                src={this.state.photo}
                className="border p-2"
                style={{ height: 500 + "px" }}
              />
            </div>
          </div>
        </div>

        {this.state.breedToGuess !== "" ? (
          <GuessButtons breedToGuess={this.state.breedToGuess} />
        ) : (
          "Loading..."
        )}
      </div>
    );
  }
}

export default DogGuessr;
