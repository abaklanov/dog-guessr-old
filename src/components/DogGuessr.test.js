import React from "react";
import { shallow } from "enzyme";
import DogGuessr from "./DogGuessr";
import axios from "axios";

const wrapper = shallow(<DogGuessr />);

// Setup
const mockData = {
  status: "success",
  message: "https://images.dog.ceo/breeds/samoyed/n02111889_6622.jpg"
};

jest.mock("axios", () => {
  return {
    get: () =>
      Promise.resolve({
        status: 200,
        data: {
          status: "success",
          message: "https://images.dog.ceo/breeds/samoyed/n02111889_6622.jpg"
        }
      })
  };
});

it("renders correctly", () => {
  expect(wrapper).toMatchSnapshot();
});

it("retrieves a random photo from API and sets it to the state when loaded", () => {
  expect(wrapper.state().photo).toEqual(mockData.message);
});

it("displays a photo of the dog in a div with id `photo`", () => {
  const div = wrapper.find("div#photo");
  expect(div).toHaveLength(1);
});
