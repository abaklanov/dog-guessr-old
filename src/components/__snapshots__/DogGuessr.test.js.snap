// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="row justify-content-center"
  >
    <div
      className="col-lg-6 text-center"
    >
      <div
        className=" m-1"
        id="photo"
      >
        <img
          className="border p-2"
          src="https://images.dog.ceo/breeds/samoyed/n02111889_6622.jpg"
          style={
            Object {
              "height": "500px",
            }
          }
        />
      </div>
    </div>
  </div>
  <GuessButtons
    breedToGuess="samoyed"
  />
</div>
`;
